### DISCUSSION TITLE
Tdarr LXC on Intel i7-1360P - QSV not working

### DISCUSSION THREAD

**Discussion by damianharbr** [tteck/Proxmox#1908] [Help Needed]
I managed to get Tdarr working fine on a 12th gen intel with QSV in a VM, but I had no luck with passing through the Intel Iris on the 13th gen machine, so I tried this script.

The Plex LXC I tested and it hardware transcoding works, so I know that it's possible to pass it through on a gen 13. 
But I can't get Tdarr QSV working with any of the plugins. 

The regular one I use is throwing up this error

`

Device creation failed: -12.

Failed to set value 'qsv:hw_any,child_device_type=vaapi' for option 'init_hw_device': Cannot allocate memory

Error parsing global options: Cannot allocate memory`

I'm not super knowledgable about how this all works, so if anyone has any suggestions for me it would be appreciated. The plugin in question is Boosh Transcode QSV which I've had success with in the past on older machines.

**Comment by dennis-vanstrien** [comment]
Hello i'm facing the same problem, did you managed to solved this? 
Failed to set value 'qsv:hw_any,child_device_type=vaapi' for option 'init_hw_device': Cannot allocate memory
Error parsing global options: Cannot allocate memory

**Comment by kraizelburg** [comment]
Same error as above, GPU is correctly passed through as it works in jellyfin with transcoding just fine.

**Comment by kaitallaoua** [comment]
Same issue on a i5-8500T, both unprivileged and privileged have the same issue, using the same plugin

**Comment by kaitallaoua** [comment]
I found the issue for my cpu at least. seems some confusion on intel-media-va-driver-non-free, my cpu needs it, only intel-media-va-driver states: 

> Only a limited set of encoders is available via this driver: JPEG (Skylake and newer), AVC (Boxton and newer), HEVC and VP9 (Ice Lake and newer

.. and im using coffee lake. Also, the ffmpeg included with tdarr does not have qsv built in which is suprising. Need to use another one. Check with `/opt/tdarr/Tdarr_Node/node_modules/ffmpeg-static/ffmpeg -encoders -hide_banner | grep qsv` nothing shows up = bad

https://packages.debian.org/bullseye/intel-media-va-driver
https://packages.debian.org/bookworm/intel-media-va-driver-non-free
https://github.com/tteck/Proxmox/discussions/1278
https://github.com/BtbN/FFmpeg-Builds/releases/tag/latest

Slightly incomplete/lazy commands: 
---> if your cpu is new enough you may not need to install the non-free drivers.. idk
```
cat <<EOF >/etc/apt/sources.list

deb http://deb.debian.org/debian bookworm main contrib non-free non-free-firmware
deb-src http://deb.debian.org/debian bookworm main contrib non-free non-free-firmware

deb http://deb.debian.org/debian-security bookworm-security main contrib non-free non-free-firmware
deb-src http://deb.debian.org/debian-security bookworm-security main contrib non-free non-free-firmware

deb http://deb.debian.org/debian bookworm-updates main contrib non-free non-free-firmware
deb-src http://deb.debian.org/debian bookworm-updates main contrib non-free non-free-firmware
EOF

apt-get update
apt-get install -y intel-media-va-driver-non-free
```

Pick your arch for ffmpeg

A good guess is: ` wget https://github.com/BtbN/FFmpeg-Builds/releases/download/latest/ffmpeg-n7.1-latest-linux64-gpl-7.1.tar.xz`

`tar -xvf ffmpeg-n7.1-latest-linux64-gpl-7.1.tar.xz`

`mv` the `ffmpeg-n7.1-latest-linux64-gpl-7.1` wherever. I chose `/opt/tdarr/Tdarr_Node/node_modules/ffmpeg-static-7.1/` to keep old ffmpeg if needed

sanity check this ffmpeg was built with qsv support: `/opt/tdarr/Tdarr_Node/node_modules/ffmpeg-static-7.1/bin/ffmpeg -encoders -hide_banner | grep qsv`

should show similar to:
```
 V..... av1_qsv              AV1 (Intel Quick Sync Video acceleration) (codec av1)
 V..... h264_qsv             H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 (Intel Quick Sync Video acceleration) (codec h264)
 V..... hevc_qsv             HEVC (Intel Quick Sync Video acceleration) (codec hevc)
 V..... mjpeg_qsv            MJPEG (Intel Quick Sync Video acceleration) (codec mjpeg)
 V..... mpeg2_qsv            MPEG-2 video (Intel Quick Sync Video acceleration) (codec mpeg2video)
 V..... vp9_qsv              VP9 video (Intel Quick Sync Video acceleration) (codec vp9)
```

`nano /opt/tdarr/configs/Tdarr_Node_Config.json`

change the ffmpeg path to your new one:

`"ffmpegPath": "/opt/tdarr/Tdarr_Node/node_modules/ffmpeg-static-7.1/bin/ffmpeg",`

`rm ffmpeg-n7.1-latest-linux64-gpl-7.1.tar.xz`

reboot ct

`reboot`

**Reply by nfribeiro** [reply]
Thank You so much. You solved mine also.  Intel i5-10400 CPU

**Reply by Titan1732** [reply]
Thanks! This solved it for my i5-8500t

EDIT:
This version of FFMPEG was badly breaking the transcoded videos - transcoded file playback was extremely artifacty and glitchy/not playable.

I instead used [jellyfin-ffmpeg](https://github.com/jellyfin/jellyfin-ffmpeg/releases) as described [here by tteck](https://github.com/tteck/Proxmox/discussions/1765#discussioncomment-7764610) and now my transcodes are working perfectly using the "Boosh-Transcode Using QSV GPU & FFMPEG" tdarr plugin

TL:DR
I got the latest version of jellyfin-ffmpeg
`wget https://github.com/jellyfin/jellyfin-ffmpeg/releases/download/v7.0.2-4/jellyfin-ffmpeg7_7.0.2-4-bookworm_amd64.deb`

Had to install a missing dependency: 
`apt install libllvm16`

installed jellyfin-ffmpeg
`dpkg -i jellyfin-ffmpeg7_7.0.2-4-bookworm_amd64.deb`

changed the node config to use this version of ffmpeg instead
`nano /opt/tdarr/configs/Tdarr_Node_Config.json`
`"ffmpegPath": "/usr/lib/jellyfin-ffmpeg/ffmpeg",`
